recipes/faiss/meta.yaml

```diff
@@ -0,0 +1,82 @@
+{% set version = "1.6.3" %}
+
+package:
+  name: faiss
+  version: {{ version }}
+
+source:
+  url: https://github.com/facebookresearch/faiss/archive/v{{ version }}.tar.gz
+  sha256: e1a41c159f0b896975fbb133e0240a233af5c9286c09a28fde6aefff5336e542
+
+build:
+  number: 0
+  skip: True  # [win]
+
+# Need these up here for conda-smithy to handle them properly.
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+  host:
+    - python
+  run:
+    - python
+
+outputs:
+  - name: faiss-cpu
```

Comment:
Would it makes to split the C++ bits and Python bits into 2 separate packages?

Reply:
Since the C++ bits get configured differently for CPU/GPU, it seems to me that this would be a bit redundant? I'm happy to be wrong about that though.

Reply:
The CPU version is simply a trimmed down version without the GPU indexes, so the CPU part remains the same in all packages, and it is indeed possible to separate the two.

Reply:
We could also use a mutex package to allow users to select between the CPU-only and GPU builds. Though certainly having everything in an all-in-one package is nicer 🙂

Reply:
So what is `faiss-cpu` containing in this?

Reply:
> So what is faiss-cpu containing in this?

IIUC, the C++ lib (without the GPU indexes?) + some python bindings. Could you comment @beauby (also in the context of the request of people to have cpp-only packaging)?

Or maybe we need two of each?
```
faiss-cpu-cpp
faiss-cpu
faiss-gpu-cpp
faiss-gpu
```

Reply:
So I looked at the size of the packages from the existing packages in the pytorch channel, and the size difference (which IIUC is just driven by the GPU indexes) is quite substantial. It's 2.1MB for [`faiss-cpu`](https://anaconda.org/pytorch/faiss-cpu/files) and 34.2MB for [`faiss-gpu`](https://anaconda.org/pytorch/faiss-gpu/files). Both of those contain the python bindings as well.

So while the build is very similar for all of them, I'm guessing it *would* make sense to package the four flavours separately (i.e. with/without python; with/without GPU indexes - cf. comment directly above)?

@beauby @mdouze What do you think?